mongo/src/main/scala/it/agilelab/darwin/connector/mongo/MongoConnector.scala

```diff
@@ -1,19 +1,165 @@
+  //    schema:    schema           (string)
+  //    name:      schema.name      (string)
+  //    namespace: schema.namespace (string)
+
+  /**
+   * return the MongoClient
+   * @param mongoConf : config with keys defined by ConfigurationKeys object
+   * @return MongoClient
+   */
+  private def createConnection(mongoConf: MongoConfig): MongoClient = {
+
+    val credential: MongoCredential = createCredential(mongoConf.user, mongoConf.db, mongoConf.password.toCharArray)
+
+    val hosts: List[ServerAddress] = mongoConf.hosts.map(host => new ServerAddress(host))
+
+    val settings: MongoClientSettings = MongoClientSettings.builder()
+      .credential(credential)
+      .applyToClusterSettings(builder => builder.hosts(hosts.asJava))
+      //.streamFactoryFactory(NettyStreamFactoryFactory())
+      .build()
+
+    MongoClient(settings)
+
+  }
+
+  private def parser: Parser = new Parser()
+
+  override def fullLoad(): Seq[(Long, Schema)] = {
+    log.debug(s"loading all schemas from table ${mongoConfig.table}")
+    val collectionNames =
+      mongoClient
+        .getDatabase(mongoConfig.db)
+        .listCollectionNames()
+
+    val schemas: Seq[(Long, Schema)] =
+      Await.result(
+        collectionNames.flatMap(collectionName =>
+          mongoClient.getDatabase(mongoConfig.db).getCollection(collectionName).find().map(document => {
+            val key = document.filterKeys(k => k == "id").head._2.asInt64().getValue
+            val schema = parser.parse(document.filterKeys(k => k == "schema").head._2.asString().getValue)
+            key -> schema
+          })
+        ).toFuture()
+        ,
+        Duration.Undefined
+      )
+
+    log.debug(s"${schemas.size} loaded from MongoDB")
+    schemas
+  }
+
+  override def insert(schemas: Seq[(Long, Schema)]): Unit = {
+
+    schemas.foreach{
+      case (id, schema) => {
+        val document = new BsonDocument
+        document.put("id", bson.BsonInt64(id))
+        document.put("schema", bson.BsonString(schema.toString))
+        document.put("name", bson.BsonString(schema.getName))
+        document.put("namespace", bson.BsonString(schema.getNamespace))
+
+        mongoClient.getDatabase(mongoConfig.db).getCollection(mongoConfig.table).insertOne(document)
+      }
+    }
+
+  }
+
+  override def createTable(): Unit = {
+    log.debug(s"Creating table ${mongoConfig.table}")
+    mongoClient.getDatabase(mongoConfig.db).createCollection(mongoConfig.table)
+    log.debug(s"Table ${mongoConfig.table} created")
+  }
+
+  override def tableExists(): Boolean = {
+    Await.result(
+      mongoClient
+        .getDatabase(mongoConfig.db)
+        .listCollectionNames()
+        .filter(x => x == mongoConfig.table)
+        .toFuture().map(_.size)
+      ,
+      Duration.Undefined
+    ) == 1
+  }
+
+  override def tableCreationHint(): String = {
+    s"""To create table perform the following command:
+       |mongoClient.getDatabase(<DATABASE>).createCollection(<TABLE>)
```

Comment:
I would put here a CLI command, if we do not know It we can also return the empty string